Modules/faulthandler.c

```diff
@@ -212,6 +216,81 @@ faulthandler_dump_traceback(int fd, int all_threads,
     reentrant = 0;
 }
 
+#ifdef HAVE_EXECINFO_H
+static void
+faulthandler_stack_dump_impl(int fd)
+{
+#define SIZE 128
+#define ENTRY_SIZE 256
+    void *callstack[SIZE];
+    int frames = backtrace(callstack, SIZE);
+    char **strings = backtrace_symbols(callstack, SIZE);
+    if (strings == NULL)
+    {
+        PUTS(fd, "  <failed to get stack trace>");
+    }
+    else {
+        for (int i = 0; i < frames; ++i)
+        {
+            char entry_str[ENTRY_SIZE];
+            snprintf(entry_str, ENTRY_SIZE, "  %s\n", strings[i]);
```

Comment:
Do we generally use the native snprintf or do we use a wrapper for this one?

Reply:
No idea. We have to be careful when using any Python-level functions because things can be corrupted.